blockmanager.go

```diff
@@ -480,8 +510,9 @@ func (b *blockManager) handleTxMsg(tmsg *txMsg) {
 	// instances of trying to fetch it, or we failed to insert and thus
 	// we'll retry next time we get an inv.
 	txHash := tmsg.tx.Sha()
-	delete(tmsg.peer.requestedTxns, *txHash)
-	delete(b.requestedTxns, *txHash)
+	for _, p := range b.peers {
```

Comment:
This should only be deleting from the specific peer that requested it, not all of them.